### PR TITLE
feat(profiles): implement removing of profiles

### PR DESCRIPTION
**Pull request by petrjasek**
- if profile is used it can't be removed
- if profile is removed also all templates which are using it are
  removed

SD-3895

### CODE REVIEW

**Comment by akintolga** [comment]
Instead of creating an in_use field, wouldn't it be better to query archive in elastic if any story with content_profile_id exists? Imo with this way it would be harder to maintain the cases like expired stories where in_use needs to revert to false if there's no other story.

File: apps/content_types/content_types.py [hunk 1]
```diff
@@ -114,5 +119,21 @@ def on_create(self, docs):
             self._set_updated_by(doc)
             self._set_created_by(doc)
 
+    def on_delete(self, doc):
+        if doc.get('is_used'):
+            raise SuperdeskApiError(status_code=202, payload={"is_used": True})
+
+    def on_deleted(self, doc):
+        superdesk.get_resource_service('content_templates').delete({'data.profile': doc.get('_id')})
```
Comment (mdhaman) [inline apps/content_types/content_types.py]: Why do we have to delete all templates if profile is deleted?

Comment (petrjasek) [inline apps/content_types/content_types.py]: good question, ping @hlmnrmr

**Comment by petrjasek** [comment]
@akintolga item could be in legal archive or it could be used by one version of an item, and I didn't want to query everything. maybe it should let admin remove it anyhow but show an extra warning if this flag is on

**Comment by akintolga** [comment]
:+1:  Sounds good.

**Comment by mdhaman** [comment]
:+1:

**Comment by petrjasek** [comment]
removed that remove all templates using it part

**Comment by mugurrus** [comment]
:+1:

**Pull request merged**